bigbio/hub/hub_repos/chebi/chebi.py

```diff
@@ -0,0 +1,247 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import datasets
+from lxml import etree
+
+from .bigbiohub import BigBioConfig, Tasks, kb_features
+
+_LANGUAGES = ["English"]
+_PUBMED = False
+_LOCAL = False
+
+_CITATION = """\
+@article{,
+title = {ChEBI: a database and ontology for chemical entities of biological interest},
+author = {Degtyarenko, Kirill and de Matos, Paula and Ennis, Marcus and Hastings, Janna and Zbinden, Martin and \
+    McNaught, Alan and Alcántara, Rafael and Darsow, Michael and Guedj, Mickaël and Ashburner, Michael},
+doi = {10.1093/nar/gkm791},
+number = {Database issue},
+volume = {36},
+month = {January},
+year = {2008},
+journal = {Nucleic acids research},
+issn = {0305-1048},
+pages = {D344—50},
+url = {https://europepmc.org/articles/PMC2238832},
+biburl = {https://aclanthology.org/W19-5008.bib},
+bibsource = {https://aclanthology.org/W19-5008/}
+}
+"""
+
+_DATASETNAME = "chebi"
+_DISPLAYNAME = "Chebi"
+
+_DESCRIPTION = """\
+ChEBI Chapti contains the results of a collaboration between the European Patent Office and
+the ChEBI team. The goal of the project was to identify chemicals within patents and cross-
+reference them to ChEBI. The teams manually annotated chemicals in a set of 40 patents.
+This was used to measure the performance of the various text-mining tools. This set of
+40 patents is distributed in this directory. The results of this work can be seen on the ChEBI
+website.
+"""
+
+_HOMEPAGE = "https://sourceforge.net/projects/chebi/"
+_LICENSE = "CC_BY_SA_4p0"
+
+DATA_URL = "https://github.com/bigscience-workshop/biomedical/files/8568960/PatentAnnotations_GoldStandard.tar.gz"
+_URLS = {
+    # The original dataset is hosted on CVS on sourceforge. Hence I have downloaded and reuploded it as tar.gz format.
```

Comment:
The provenance of this dataset seems to be tricky. It is not mentioned in the original ChEBI publication in NAR, that we have as a citation. Is there any other source of information about the annotation project? Otherwise, it is a bit hard to tell if we got the number of annotations, IDs, etc. right.